src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java

```diff
@@ -0,0 +1,185 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package com.vesoft.nebula.graph.client;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.facebook.thrift.TException;
+import com.facebook.thrift.protocol.TBinaryProtocol;
+import com.facebook.thrift.protocol.TProtocol;
+import com.facebook.thrift.transport.TSocket;
+import com.facebook.thrift.transport.TTransport;
+import com.facebook.thrift.transport.TTransportException;
+import com.google.common.collect.Lists;
+import com.google.common.net.InetAddresses;
+import com.vesoft.nebula.graph.AuthResponse;
+import com.vesoft.nebula.graph.ErrorCode;
+import com.vesoft.nebula.graph.ExecutionResponse;
+import com.vesoft.nebula.graph.GraphService;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GraphClientImpl implements GraphClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GraphClientImpl.class.getName());
+
+    private int retry;
+    private final int timeout;
+    private long sessionId_;
+    private TTransport transport = null;
+    private GraphService.Client client;
+    private List<InetSocketAddress> nodes = Lists.newLinkedList();
+
+    public GraphClientImpl(Map<String, Integer> addresses, int timeout, int retry) {
```

Comment:
If multiple instances with different ports are running on the same ip (Ps, ip1:3307, ip1:3308), the `addresses` map can only save one.

Reply:
That's strange.